shub/settings.py

```diff
@@ -0,0 +1,662 @@
+    "SOCIAL_AUTH_GLOBUS_SECRET": None,
+    "GLOBUS_ENDPOINT_ID": None,
+    # SAML Authentication (saml)
+    # Only required if 'saml_auth' is added to PLUGINS_ENABLED
+    "AUTH_SAML_IDP": None,  # "stanford"
+    "AUTH_SAML_INSTITUTION": None,  # "Stanford University"
+}
+
+INTEGER_DEFAULTS = {
+    "API_PAGE_SIZE": 10,
+    # Set this to be some size in MB to limit uploads.
+    # Uploads > 2.5GB will not use memory, but the filesystem
+    "DATA_UPLOAD_MAX_MEMORY_SIZE": None,
+    # Limit users to N collections (None is unlimited)
+    "USER_COLLECTION_LIMIT": 2,
+    # The number of collections to show on the /<domain>/collections page
+    "COLLECTIONS_VIEW_PAGE_COUNT": 250,
+    # The maximum number of downloads allowed per container/collection, per week
+    "CONTAINER_WEEKLY_GET_LIMIT": 100,
+    "COLLECTION_WEEKLY_GET_LIMIT": 100,
+    "GOOGLE_BUILD_LIMIT": 100,
+    "GOOGLE_BUILD_TIMEOUT_SECONDS": None,  # None defaults to 10 minutes
+    "GOOGLE_BUILD_EXPIRE_SECONDS": 28800,
+    "MINIO_SIGNED_URL_EXPIRE_MINUTES": 5,
+    # Google Build
+    # To prevent denial of service attacks on Google Cloud Storage, you should set a reasonable limit for the number of active, concurrent builds.
+    # This number should be based on your expected number of users, repositories, and recipes per repository.
+    "GOOGLE_BUILD_LIMIT": None,  # 100
+    # The number of seconds for the build to timeout. If set to None, will be 10 minutes. If
+    # unset, will default to 3 hours. This time should be less than the GOOGLE_BUILD_EXPIRE_SECONDS
+    "GOOGLE_BUILD_TIMEOUT_SECONDS": None,
+    # The number of seconds for the build to expire, meaning it's response is no longer accepted by the server. This must be defined.
+    # Using 28800 would indicate 8 hours (in seconds)
+    "GOOGLE_BUILD_EXPIRE_SECONDS": None,  # 28800
+    # The number of seconds to expire a signed URL given to download a container
+    # from storage. This can be much smaller than 10, as we only need it to endure
+    # for the POST.
+    "CONTAINER_SIGNED_URL_EXPIRE_SECONDS": None,  # 10
+}
+
+# Environment helpers
+
+
+def get_sregistry_envar(key):
+    envar_key = "SREGISTRY_%s" % key
+    return os.environ.get(envar_key)
+
+
+def get_sregistry_envar_list(key):
+    key = "SREGISTRY_%s" % key
+    envar_list = os.environ.get(key) or []
+    if envar_list:
+        envar_list = [x.strip() for x in envar_list.split(",") if x.strip()]
+    return envar_list
+
+
+# Try reading in from secrets first (no issue if not found)
+try:
+    from .secrets import *
+except ImportError:
+    pass
+
+# Set boolean defaults from environment
+for key in BOOLEAN_DEFAULTS:
+    value = get_sregistry_envar(key)
+
+    if value is None:
+        continue
+
+    # Setting a boolean
+    if value.lower() in ["true", "1"]:
+        BOOLEAN_DEFAULTS[key] = True
+
+    elif value.lower() in ["false", "1"]:
```

Comment:
1 should be 0 here.

Reply:
haha oops! I'll update these in a few minutes - finishing up something else!